### ISSUE TITLE
Squiz.Commenting.FunctionComment.MissingReturn @inheritDoc not resolved?

### ISSUE THREAD

**Issue by kwhat** [squizlabs/PHP_CodeSniffer#2770]
Hi,

I have `@inheritDoc` as the docblock to one of my inherited methods.  The parent method is documented correctly, however, Squiz.Commenting does not appear to inherit the doc-block.

```
    /**
     * @inheritDoc
     */
    public function __toString(): string { /* Child Class */ }
```

```
    /**
     * @throws DatabaseException
     *
     * @return string
     */
    public function __toString(): string { /* Parent Class */ }
```

```
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 1 LINE
----------------------------------------------------------------------
 31 | ERROR | Missing @return tag in function comment
    |       | (Squiz.Commenting.FunctionComment.MissingReturn)
 31 | ERROR | Missing @throws tag in function comment
    |       | (Squiz.Commenting.FunctionCommentThrowTag.Missing)
----------------------------------------------------------------------
```

**Comment by kwhat** [comment]
Tested with `{@inheritdoc}` and `{@inheritDoc}`, Possible regression of #214

**Comment by kwhat** [comment]
Never mind, you learn something new every day.

**Comment by bkdotcom** [comment]
Not sure why this was closed.   There's currently no open issue regarding inheritdoc

Seems like like if the sniff had a `$skipIfInheritdoc` (default `false`) option, everyone would be happy

`$skipIfInheritdoc` would address the common use case

```php
    /**
     * {@inheritdoc}
     */
```

**Comment by kwhat** [comment]
I closed this issue after reading the PHP doc documentation, CS works according to the specification as it currently is. @inheritDoc is only for the title and description.

**Comment by bkdotcom** [comment]
https://docs.phpdoc.org/latest/guides/inheritance.html

> However, it does make clear that an element has been explicitly documented (and thus not forgotten). As such we are working to include a new (normal) tag in the PHPDoc Standard @inheritDoc that will serve that purpose.

8 years have passed...

That may be the "official" specification, but the community has adopted it to include the the entire parent comment if the comment consists only of @inheritdoc

eclipse, PhpStorm, Symfony for starters support/ use it for this purpose

**Comment by filips123** [comment]
Can this be reopened and supported? Basically any documentation builder (including phpDocumentor), framework and modern IDE supports `@inheritDoc` and `{@inheritDoc}`.

Also, maybe a bit newer standard is [PSR-19](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc-tags.md) draft which specified:

> In addition to the inherited descriptions and tags as defined in this chapter's root, a function or method in a class or interface MUST inherit the following tags:
>
> * `@param`
> * `@return`
> * `@throws`

**Comment by xjm** [comment]
A `$skipIfInheritdoc` option seems like a great solution to this, given that this feature request has been made over and over for years on various docblock-related sniffs.

**Comment by kwhat** [comment]
Reopened the issue for clarification from devs.

**Comment by xjm** [comment]
#3051 solves this for me locally. Example usage for testing:
```
    <rule ref="Squiz.Commenting.FunctionComment">                               
        <properties>
            <property name="skipIfInheritdoc" value="true"/>
        </properties>
    </rule>
```

**Comment by gsherwood** [comment]
This has now been resolved by the sniff property added in #3051, and documented here: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Customisable-Sniff-Properties#squizcommentingfunctioncomment

**Comment by XedinUnknown** [comment]
Not working. Even with `skipIfInheritDoc` still reports `Squiz.Commenting.FunctionComment.MissingParamTag`.

**Comment by gsherwood** [comment]
> [Not working](https://github.com/XedinUnknown/woocommerce-paypal-payments/pull/1/checks?check_run_id=2093816114#step:9:10) for me. Even [with](https://github.com/XedinUnknown/woocommerce-paypal-payments/blob/task/refactoring/phpcs.xml.dist#L29) `skipIfInheritDoc` still reports `Squiz.Commenting.FunctionComment.MissingParamTag`.

This new option is unreleased and not available in 3.5.8. This will become available when 3.6.0 is released.

**Comment by XedinUnknown** [comment]
Oh! I just assumed there'd be a releaase since mid-january this year. What about making a beta release with this? :)
